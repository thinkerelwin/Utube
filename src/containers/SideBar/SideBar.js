import React from 'react';
import SideBarItem from './SideBarItem/SideBarItem';
import { SideBarHeader } from './SideBarHeader/SideBarHeader';
import { Subscriptions } from './Subscriptions/Subscriptions';
import { SideBarFooter } from './SideBarFooter/SideBarFooter';
import { Menu, Divider } from 'semantic-ui-react';
import './SideBar.scss';

export const SideBar = () => (
  <Menu borderless vertical stackable fixed="left" className="side-nav">
    <SideBarItem path="/" label="Home" icon="home" />
    <SideBarItem path="/feed/trending" label="Trending" icon="fire" />
    <SideBarItem label="Followers" icon="spy" />
    <Divider />
    <SideBarItem label="Library" icon="folder" />
    <SideBarItem label="History" icon="history" />
    <SideBarItem label="Watch later" icon="clock" />
    <SideBarItem label="Liked videos" icon="thumbs up" />
    <Divider />
    <Subscriptions />
    <Divider />
    <SideBarHeader title="More from Youtube" />
    <SideBarItem label="Movies and Shows" icon="film" />
    <Divider />
    <SideBarItem label="Report history" icon="flag" />
    <SideBarItem label="Help" icon="help circle" />
    <SideBarItem label="Send feedback" icon="comment" />
    <Divider />
    <SideBarFooter />
  </Menu>
);
