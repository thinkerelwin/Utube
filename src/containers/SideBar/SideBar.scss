@import '../../styles/shared.scss';

$sidebar-hover-color: #ebebeb;

.ui.menu.fixed.side-nav {
  background-color: #f5f5f5;
  margin-top: $header-nav-height;
  padding-bottom: $header-nav-height;
  overflow-y: auto;

  .sidebar-item:hover {
    background: #ebebeb;
  }
}

.side-nav.ui.vertical.menu {
  width: 17rem;
}
