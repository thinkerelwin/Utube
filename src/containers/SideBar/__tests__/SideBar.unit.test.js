import React from 'react';
import { shallow } from 'enzyme';
import { SideBar } from '../SideBar';

test('renders SideBar', () => {
  const wrapper = shallow(<SideBar />);
  expect(wrapper).toMatchSnapshot();
});
