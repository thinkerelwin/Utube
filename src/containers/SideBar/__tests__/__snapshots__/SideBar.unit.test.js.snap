// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SideBar 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <withRouter(SideBarItem)
    icon="home"
    label="Home"
    path="/"
  />
  <withRouter(SideBarItem)
    icon="fire"
    label="Trending"
    path="/feed/trending"
  />
  <withRouter(SideBarItem)
    icon="spy"
    label="Followers"
  />
  <Divider />
  <withRouter(SideBarItem)
    icon="folder"
    label="Library"
  />
  <withRouter(SideBarItem)
    icon="history"
    label="History"
  />
  <withRouter(SideBarItem)
    icon="clock"
    label="Watch later"
  />
  <withRouter(SideBarItem)
    icon="thumbs up"
    label="Liked videos"
  />
  <Divider />
  <Subscriptions />
  <Divider />
  <SideBarHeader
    title="More from Youtube"
  />
  <withRouter(SideBarItem)
    icon="film"
    label="Movies and Shows"
  />
  <Divider />
  <withRouter(SideBarItem)
    icon="flag"
    label="Report history"
  />
  <withRouter(SideBarItem)
    icon="help circle"
    label="Help"
  />
  <withRouter(SideBarItem)
    icon="comment"
    label="Send feedback"
  />
  <Divider />
  <SideBarFooter />
</Menu>
`;
